client.go

```diff
@@ -429,113 +428,95 @@ func (c *Client) Subscribe(intv time.Duration) (*Subscription, error) {
-	Error                    error
-	SubscriptionID           uint32
-	DataChangeNotification   *ua.DataChangeNotification
-	EventNotificationList    *ua.EventNotificationList
-	StatusChangeNotification *ua.StatusChangeNotification
+	SubscriptionID uint32
+	Error          error
+	Value          interface{}
 }
 
-func (c *Client) Publish(dataChan chan<- PublishNotificationData) {
-	var (
-		subAck        = make([]*ua.SubscriptionAcknowledgement, 0)
-		pubRequest    = &ua.PublishRequest{}
-		pubResponse   *ua.PublishResponse
-		responseError ua.StatusCode
-		err           error
-	)
+func (c *Client) Publish(notif chan<- PublishNotificationData) {
 	for {
 		// Empty SubscriptionAcknowledgements for first PublishRequest
-		pubRequest.SubscriptionAcknowledgements = subAck
+		req := &ua.PublishRequest{
+			SubscriptionAcknowledgements: []*ua.SubscriptionAcknowledgement{},
+		}
 
-		// PublishRequest
-		err = c.Send(pubRequest, func(v interface{}) error {
+		var res *ua.PublishResponse
+		err := c.Send(req, func(v interface{}) error {
 			r, ok := v.(*ua.PublishResponse)
 			if !ok {
 				return fmt.Errorf("invalid response: %T", v)
 			}
-
-			pubResponse = r
+			res = r
 			return nil
 		})
 		if err != nil {
-			dataChan <- PublishNotificationData{
-				Error: err,
-			}
+			notif <- PublishNotificationData{Error: err}
 			continue
 		}
 
 		// Check for errors
-		responseError = ua.StatusOK
-		for _, result := range pubResponse.Results {
-			if result != ua.StatusOK {
-				responseError = result
+		status := ua.StatusOK
+		for _, res := range res.Results {
+			if res != ua.StatusOK {
+				status = res
 				break
 			}
 		}
-		if responseError != ua.StatusOK {
-			dataChan <- PublishNotificationData{
-				Error:          fmt.Errorf("publish response error: %v", responseError),
-				SubscriptionID: pubResponse.SubscriptionID,
+
+		if status != ua.StatusOK {
+			notif <- PublishNotificationData{
+				SubscriptionID: res.SubscriptionID,
+				Error:          status,
 			}
 			continue
 		}
 
 		// Prepare SubscriptionAcknowledgement for next PublishRequest
-		subAck = make([]*ua.SubscriptionAcknowledgement, 0)
-		for _, number := range pubResponse.AvailableSequenceNumbers {
-			subAck = append(subAck, &ua.SubscriptionAcknowledgement{
-				SubscriptionID: pubResponse.SubscriptionID,
-				SequenceNumber: number,
-			})
+		var acks []*ua.SubscriptionAcknowledgement
+		for _, i := range res.AvailableSequenceNumbers {
+			ack := &ua.SubscriptionAcknowledgement{
+				SubscriptionID: res.SubscriptionID,
+				SequenceNumber: i,
+			}
+			acks = append(acks, ack)
 		}
 
-		if pubResponse.NotificationMessage == nil {
-			dataChan <- PublishNotificationData{
-				Error:          errors.New("empty NotificationMessage"),
-				SubscriptionID: pubResponse.SubscriptionID,
+		if res.NotificationMessage == nil {
+			notif <- PublishNotificationData{
+				SubscriptionID: res.SubscriptionID,
+				Error:          fmt.Errorf("empty NotificationMessage"),
```

Comment:
`fmt.Error`?

Reply:
Simplicity here as well. The code uses `fmt.Errorf` in some places already and there is no win in using a different approach for some cases.